src/scanRegistration.cpp

```diff
@@ -832,9 +832,9 @@ void imuHandler(const sensor_msgs::Imu::ConstPtr& imuIn)
     //imuAccuPitch += timeDiff * imuIn->angular_velocity.y;
     // imuAccuYaw += timeDiff * imuIn->angular_velocity.z;
     
-    imuRoll[imuPointerLast] = roll;
-    imuPitch[imuPointerLast] = -pitch;
-    imuYaw[imuPointerLast] = -yaw;
+    imuRoll[imuPointerLast] = yaw;
```

Comment:
@gtinchev : 
I don't think this isn't correct but I have a very good idea about what is going on here.
I think you are very close to resolving this. Lets talk in the afternoon